commit f1c6872e4980bc4078cfaead05f892b3d78dea64
Merge: 5f3d2f2e1a63 4ed5978bdd99
Author: Linus Torvalds <torvalds@linux-foundation.org>
Date:   Sun Oct 7 07:13:01 2012 +0900

    Merge tag 'stable/for-linus-3.7-arm-tag' of git://git.kernel.org/pub/scm/linux/kernel/git/konrad/xen
    
    Pull ADM Xen support from Konrad Rzeszutek Wilk:
    
      Features:
       * Allow a Linux guest to boot as initial domain and as normal guests
         on Xen on ARM (specifically ARMv7 with virtualized extensions).  PV
         console, block and network frontend/backends are working.
      Bug-fixes:
       * Fix compile linux-next fallout.
       * Fix PVHVM bootup crashing.
    
      The Xen-unstable hypervisor (so will be 4.3 in a ~6 months), supports
      ARMv7 platforms.
    
      The goal in implementing this architecture is to exploit the hardware
      as much as possible.  That means use as little as possible of PV
      operations (so no PV MMU) - and use existing PV drivers for I/Os
      (network, block, console, etc).  This is similar to how PVHVM guests
      operate in X86 platform nowadays - except that on ARM there is no need
      for QEMU.  The end result is that we share a lot of the generic Xen
      drivers and infrastructure.
    
      Details on how to compile/boot/etc are available at this Wiki:
    
        http://wiki.xen.org/wiki/Xen_ARMv7_with_Virtualization_Extensions
    
      and this blog has links to a technical discussion/presentations on the
      overall architecture:
    
        http://blog.xen.org/index.php/2012/09/21/xensummit-sessions-new-pvh-virtualisation-mode-for-arm-cortex-a15arm-servers-and-x86/
    
    * tag 'stable/for-linus-3.7-arm-tag' of git://git.kernel.org/pub/scm/linux/kernel/git/konrad/xen: (21 commits)
      xen/xen_initial_domain: check that xen_start_info is initialized
      xen: mark xen_init_IRQ __init
      xen/Makefile: fix dom-y build
      arm: introduce a DTS for Xen unprivileged virtual machines
      MAINTAINERS: add myself as Xen ARM maintainer
      xen/arm: compile netback
      xen/arm: compile blkfront and blkback
      xen/arm: implement alloc/free_xenballooned_pages with alloc_pages/kfree
      xen/arm: receive Xen events on ARM
      xen/arm: initialize grant_table on ARM
      xen/arm: get privilege status
      xen/arm: introduce CONFIG_XEN on ARM
      xen: do not compile manage, balloon, pci, acpi, pcpu and cpu_hotplug on ARM
      xen/arm: Introduce xen_ulong_t for unsigned long
      xen/arm: Xen detection and shared_info page mapping
      docs: Xen ARM DT bindings
      xen/arm: empty implementation of grant_table arch specific functions
      xen/arm: sync_bitops
      xen/arm: page.h definitions
      xen/arm: hypercalls
      ...

diff --cc arch/arm/boot/dts/Makefile
index 43c084c2cd66,000000000000..4745c1f68b49
mode 100644,000000..100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@@ -1,101 -1,0 +1,102 @@@
 +ifeq ($(CONFIG_OF),y)
 +
 +dtb-$(CONFIG_ARCH_AT91) += aks-cdu.dtb \
 +	at91sam9263ek.dtb \
 +	at91sam9g20ek_2mmc.dtb \
 +	at91sam9g20ek.dtb \
 +	at91sam9g25ek.dtb \
 +	at91sam9m10g45ek.dtb \
 +	at91sam9n12ek.dtb \
 +	ethernut5.dtb \
 +	evk-pro3.dtb \
 +	kizbox.dtb \
 +	tny_a9260.dtb \
 +	tny_a9263.dtb \
 +	tny_a9g20.dtb \
 +	usb_a9260.dtb \
 +	usb_a9263.dtb \
 +	usb_a9g20.dtb
 +dtb-$(CONFIG_ARCH_BCM2835) += bcm2835-rpi-b.dtb
 +dtb-$(CONFIG_ARCH_EXYNOS) += exynos4210-origen.dtb \
 +	exynos4210-smdkv310.dtb \
 +	exynos4210-trats.dtb \
 +	exynos5250-smdk5250.dtb
 +dtb-$(CONFIG_ARCH_HIGHBANK) += highbank.dtb
 +dtb-$(CONFIG_ARCH_IMX5) += imx51-babbage.dtb \
 +	imx53-ard.dtb \
 +	imx53-evk.dtb \
 +	imx53-qsb.dtb \
 +	imx53-smd.dtb
 +dtb-$(CONFIG_SOC_IMX6Q) += imx6q-arm2.dtb \
 +	imx6q-sabrelite.dtb \
 +	imx6q-sabresd.dtb
 +dtb-$(CONFIG_ARCH_LPC32XX) += ea3250.dtb phy3250.dtb
 +dtb-$(CONFIG_ARCH_KIRKWOOD) += kirkwood-dns320.dtb \
 +	kirkwood-dns325.dtb \
 +	kirkwood-dreamplug.dtb \
 +	kirkwood-goflexnet.dtb \
 +	kirkwood-ib62x0.dtb \
 +	kirkwood-iconnect.dtb \
 +	kirkwood-lschlv2.dtb \
 +	kirkwood-lsxhl.dtb \
 +	kirkwood-ts219-6281.dtb \
 +	kirkwood-ts219-6282.dtb
 +dtb-$(CONFIG_ARCH_MSM) += msm8660-surf.dtb \
 +	msm8960-cdp.dtb
 +dtb-$(CONFIG_ARCH_MVEBU) += armada-370-db.dtb \
 +	armada-xp-db.dtb
 +dtb-$(CONFIG_ARCH_MXC) += imx51-babbage.dtb \
 +	imx53-ard.dtb \
 +	imx53-evk.dtb \
 +	imx53-qsb.dtb \
 +	imx53-smd.dtb \
 +	imx6q-arm2.dtb \
 +	imx6q-sabrelite.dtb \
 +	imx6q-sabresd.dtb
 +dtb-$(CONFIG_ARCH_MXS) += imx23-evk.dtb \
 +	imx23-olinuxino.dtb \
 +	imx23-stmp378x_devb.dtb \
 +	imx28-apx4devkit.dtb \
 +	imx28-cfa10036.dtb \
 +	imx28-cfa10049.dtb \
 +	imx28-evk.dtb \
 +	imx28-m28evk.dtb \
 +	imx28-tx28.dtb
 +dtb-$(CONFIG_ARCH_OMAP2PLUS) += omap2420-h4.dtb \
 +	omap3-beagle-xm.dtb \
 +	omap3-evm.dtb \
 +	omap3-tobi.dtb \
 +	omap4-panda.dtb \
 +	omap4-pandaES.dtb \
 +	omap4-var_som.dtb \
 +	omap4-sdp.dtb \
 +	omap5-evm.dtb
 +dtb-$(CONFIG_ARCH_PRIMA2) += prima2-evb.dtb
 +dtb-$(CONFIG_ARCH_U8500) += snowball.dtb
 +dtb-$(CONFIG_ARCH_SHMOBILE) += emev2-kzm9d.dtb \
 +	r8a7740-armadillo800eva.dtb \
 +	sh73a0-kzm9g.dtb
 +dtb-$(CONFIG_ARCH_SPEAR13XX) += spear1310-evb.dtb \
 +	spear1340-evb.dtb
 +dtb-$(CONFIG_ARCH_SPEAR3XX)+= spear300-evb.dtb \
 +	spear310-evb.dtb \
 +	spear320-evb.dtb
 +dtb-$(CONFIG_ARCH_SPEAR6XX)+= spear600-evb.dtb
 +dtb-$(CONFIG_ARCH_TEGRA) += tegra20-harmony.dtb \
 +	tegra20-medcom-wide.dtb \
 +	tegra20-paz00.dtb \
 +	tegra20-plutux.dtb \
 +	tegra20-seaboard.dtb \
 +	tegra20-tec.dtb \
 +	tegra20-trimslice.dtb \
 +	tegra20-ventana.dtb \
 +	tegra20-whistler.dtb \
 +	tegra30-cardhu-a02.dtb \
 +	tegra30-cardhu-a04.dtb
 +dtb-$(CONFIG_ARCH_VEXPRESS) += vexpress-v2p-ca5s.dtb \
 +	vexpress-v2p-ca9.dtb \
 +	vexpress-v2p-ca15-tc1.dtb \
- 	vexpress-v2p-ca15_a7.dtb
++	vexpress-v2p-ca15_a7.dtb \
++	xenvm-4.2.dtb
 +
 +endif